### PR TITLE
Add install instructions for the OPPO Watch.

### PR DESCRIPTION
**Pull request by wannaphong**
This pull request is WIP. It add install instructions for the OPPO Watch to asteroidos.org.

https://github.com/AsteroidOS/meta-smartwatch/issues/77

### CODE REVIEW

**Comment by wannaphong** [comment]
This svg is image for Instructions. oppo-2

**Comment by wannaphong** [comment]
This url is all image for oppo watch. https://imgur.com/a/hHKRZNZ

File: pages/install/beluga.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <div class="support-col">USB<div class="support-col-good"></div></div>
+  <div class="support-col">Heart Rate<div class="support-col-good"></div></div>
+  <div class="support-col">WLAN<div class="support-col-bad"></div></div>
+  <div class="support-col">Tilt-to-Wake<div class="support-col-good"></div></div>
+  <div class="support-col">NFC<div class="support-col-bad"></div></div>
+</div>
+
+<div class="callout callout-warning">
+    <h4>Warning!</h4>
+    <p>OPPO Watch Free are not supported! See the list on the <a href="{{rel 'wiki/porting-status'}}">porting status</a> page to find out if your model is supported.</p>
```
Comment (MagneFire) [inline pages/install/beluga.md:26]: Can you change this to: `The OPPO Watch Free is not supported!`?

File: templates/layouts/beluga-install.hbs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          Then, Boot the temporary image (using the on screen boot menu). The bootlogo will popup with the message "<b>Push asteroidos.ext4 via ADB</b>" will show below the bootlogo.
+          <br>Push AsteroidOS to the internal sdcard using Linux:
+          <pre><code class="bash">adb push -p ~/Downloads/asteroid-image-{{deviceName}}.ext4 /sdcard/media/0/asteroidos.ext4</code></pre>
+
+          or using Windows:
+          <pre><code class="bash">adb push -p %systemdrive%%homepath%\Downloads\asteroid-image-{{deviceName}}.ext4 /sdcard/media/0/asteroidos.ext4</code></pre>
+
+          Everytime you want to reboot AsteroidOS from Wear OS, run:
+          <pre><code>adb reboot bootloader</code></pre>
+          Then, if you are using Linux:
```
Comment (MagneFire) [inline templates/layouts/beluga-install.hbs:98]: This part and below is no longer needed since you already installed the `fastboot` image to the recovery partition.
Instead, rephrase it to:
Then, use the top button on the watch to select "Recovery mode" and push the bottom button to boot into AsteroidOS.

File: templates/layouts/beluga-install.hbs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+fastboot continue</code></pre>
+
+          <b><li type="a">Temporary installation</li></b>
+          <b>Important to not boot into WearOS after unlock your bootloader!</b>
+          While your watch is in bootloader mode, flash the recovery partitions using Linux:
+          <pre><code>fastboot flash recovery ~/Downloads/zImage-dtb-{{deviceName}}.fastboot</code></pre>
+
+          or on Windows:
+          <pre><code>fastboot flash recovery %systemdrive%%homepath%\Downloads\zImage-dtb-{{deviceName}}.fastboot</code></pre>
+          Then, Boot the temporary image (using the on screen boot menu). The bootlogo will popup with the message "<b>Push asteroidos.ext4 via ADB</b>" will show below the bootlogo.
```
Comment (MagneFire) [inline templates/layouts/beluga-install.hbs:89]: It isn't clear how one would boot into the temporary image.
Maybe add here that one has to select recovery mode and boot to that:
While in recovery mode use the top button on the watch to select "Recovery mode" and push the bottom button to boot into the temporary image of AsteroidOS.

**Review by MagneFire — CHANGES_REQUESTED** [review]
In general this looks good :+1: 
Only some minor comments here.
Additionally, can you add the watch to the `install.md` (https://github.com/AsteroidOS/asteroidos.org/blob/master/pages/main/install.md)?

I think giving it 2 or 3 stars makes sense for now. Putting it between the Asus Zenwatch 1 and TicWatch Pro 2018/2020. Or TicWatch C2/C2+ and Ticwatch E & S.

**Comment by wannaphong** [comment]
@MagneFire Thank you! I was update the docs by your comments.

File: templates/layouts/beluga-install.hbs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+fastboot flash boot ~/Downloads/zImage-dtb-{{deviceName}}.fastboot
+fastboot continue</code></pre>
+
+          or using Windows:
+          <pre><code class="bash">fastboot flash userdata %systemdrive%%homepath%\Downloads\asteroid-image-{{deviceName}}.ext4
+fastboot flash boot %systemdrive%%homepath%\Downloads\zImage-dtb-{{deviceName}}.fastboot
+fastboot continue</code></pre>
+
+          <b><li type="a">Temporary installation</li></b>
+          <b>Important to not boot into WearOS after unlock your bootloader!</b>
```
[the remark this answers is not in the document]
Comment (wannaphong) [reply templates/layouts/beluga-install.hbs]: > It's important to not boot into WearOS after unlocking your bootloader"

Thank you! I was rewritten this sentence.

**Comment by MagneFire** [comment]
We now have official nightlies available: https://release.asteroidos.org/nightlies/beluga/

**Review by MagneFire — APPROVED** [review]

**Comment by nelikp222** [comment]
> Thanks for this!
> 
> Before merging this I would first like to open a PR that simplifies some of the setup process. Essentially the process would become:
> 
> * Enable ADB mode
> * Enable OEM unlocking
> * `adb reboot bootloader`
> * `fastboot oem unlock`
> * It will factory reset. Immediately after the factory reset DO NOT let it boot into WearOS.
>   
>   * Booting into WearOS will format the `userdata` partition to the "F2FS" file system WITH encryption enabled. This means that we cannot `adb push` the `ext4` rootfs.
>   * I'm currently looking into providing a way to format the `userdata` partition via the ramdisk. (I can already achieve this but WearOS fails to boot probably because of incorrect filesystem features.)
> * Flash the `zImage-dtb-beluga.fastboot` to either `boot` or `recovery` (real or ~temporary~ dual boot install)
> * (For the real install only): flash the userdata partition with the `asteroid-image-beluga.ext4` image.
> * (For the temporary install):
>   
>   * Boot the temporary image (using the on screen boot menu). Important to not boot into WearOS!
>   * The bootlogo will popup with the message "Push asteroidos.ext4 via ADB" will show below the bootlogo (PR soon).
>   * `adb push -p asteroid-image-beluga.ext4 /sdcard/media/0/asteroidos.ext4`
>   * Reboot the watch using: `adb reboot`

works well but for the real install
what is user partition is?
fastboot flash xxxx asteroid-image-beluga.ext4
thank you

**Comment by nelikp222** [comment]
Screenshot 2022-07-17 204106
2 more question
how to rid from this issue
need stock boot/recovery.img and system.img

**Comment by wannaphong** [comment]
> Screenshot 2022-07-17 204106 2 more question how to rid from this issue need stock boot/recovery.img and system.img

Do you use `fastboot oem unlock` before boot `boot.img`?

**Comment by wannaphong** [comment]
> 
> 
> > Thanks for this!
> > Before merging this I would first like to open a PR that simplifies some of the setup process. Essentially the process would become:
> > 
> > * Enable ADB mode
> > * Enable OEM unlocking
> > * `adb reboot bootloader`
> > * `fastboot oem unlock`
> > * It will factory reset. Immediately after the factory reset DO NOT let it boot into WearOS.
> >   
> >   * Booting into WearOS will format the `userdata` partition to the "F2FS" file system WITH encryption enabled. This means that we cannot `adb push` the `ext4` rootfs.
> >   * I'm currently looking into providing a way to format the `userdata` partition via the ramdisk. (I can already achieve this but WearOS fails to boot probably because of incorrect filesystem features.)
> > * Flash the `zImage-dtb-beluga.fastboot` to either `boot` or `recovery` (real or ~temporary~ dual boot install)
> > * (For the real install only): flash the userdata partition with the `asteroid-image-beluga.ext4` image.
> > * (For the temporary install):
> >   
> >   * Boot the temporary image (using the on screen boot menu). Important to not boot into WearOS!
> >   * The bootlogo will popup with the message "Push asteroidos.ext4 via ADB" will show below the bootlogo (PR soon).
> >   * `adb push -p asteroid-image-beluga.ext4 /sdcard/media/0/asteroidos.ext4`
> >   * Reboot the watch using: `adb reboot`
> 
> works well but for the real install what is user partition is? fastboot flash xxxx asteroid-image-beluga.ext4 thank you
> 
> 

The user partition is `userdata`. You can read at https://asteroidos.org/install/beluga/.

**Comment by nelikp222** [comment]
ok nice
i used fastboot oem unlock
but i cant adb push the file 
(i guess wrong directory issue)
when i try push to /sdcard/media/asteroidos.ext4 
(instead /sdcard/media/0/asteroidos.ext4)
file able to push but wont boot into asteroidos

when i push to /mnt/media/0/asteroidos.ext4
it like too slow at 95% transfering
and screen went to sleep

**Comment by nelikp222** [comment]
Screenshot 2022-07-17 204106

did u have stock img?
i cant boot my watch in other way
stuck on the custom boot logo
or u make the twrp then i can erase userdata or something

**Comment by wannaphong** [comment]
Can you delete `/sdcard/media/asteroidos.ext4`?

```
adb shell
rm /sdcard/media/asteroidos.ext4
```

**Comment by nelikp222** [comment]
sh-5.1# adb shell
card/media/asteroidos.ext4error: closed
sh-5.1# rm /sdcard/media/asteroidos.ext4
rm: can't remove '/sdcard/media/asteroidos.ext4': No such file or directory
sh-5.1#

**Comment by wannaphong** [comment]
OK. Can you give me the output?

`ls /sdcard/media/0/`

**Comment by nelikp222** [comment]
C:\Users\Kirt\Desktop\oppo>adb push -p asteroid-image-beluga.ext4 /mnt/media/0/asteroidos.ext4
^C 3%] /mnt/media/0/asteroidos.ext4
C:\Users\Kirt\Desktop\oppo>ls /sdcard/media/0/
'ls' is not recognized as an internal or external command,
operable program or batch file.

**Comment by nelikp222** [comment]
in this situation
how to pull system.img boot.img

**Comment by eLtMosen** [comment]
> C:\Users\Kirt\Desktop\oppo>adb push -p asteroid-image-beluga.ext4 /mnt/media/0/asteroidos.ext4 ^C 3%] /mnt/media/0/asteroidos.ext4 C:\Users\Kirt\Desktop\oppo>ls /sdcard/media/0/ 'ls' is not recognized as an internal or external command, operable program or batch file.

Its `adb shell "ls /sdcard/media/0/"` to relay the command to the watch via adb.
Else your windows tries to do `ls`, which is not available on windows.

**Comment by nelikp222** [comment]
C:\Users\Kirt\Desktop\oppo>adb push -p asteroid-image-beluga.ext4 /mnt/media/0/asteroidos.ext4
^C13%] /mnt/media/0/asteroidos.ext4
C:\Users\Kirt\Desktop\oppo>adb shell "ls /sdcard/media/0/"
,QrcTN2T3gOm67Czjlm5ED  H7ZBAz2Vrv4PmXhotNnV9B  SVNkFmYzvDD+Up4Ni3ziKB
3mNoj4xJf6AjIrO1JmgftC  HVrMmrDTQK11Axt43XvyDB  dvXvKM5TGqzWV3t55QzspC
AHKIeZH5Kx4A11QKLGYJIB  NGI,UXAgEQTgn,1ZUdwftD  l9f4jtJc7tl5AvVwJbcyhA
EyZasA+oPP5u7sqoiEZ0QC  S,7VTJqx3sikoqlYSc4VWB  wl4DbxMKdlwpluiByxtjPA

**Comment by nelikp222** [comment]
Screenshot 2022-07-17 221059

**Comment by nelikp222** [comment]
> > C:\Users\Kirt\Desktop\oppo>adb push -p asteroid-image-beluga.ext4 /mnt/media/0/asteroidos.ext4 ^C 3%] /mnt/media/0/asteroidos.ext4 C:\Users\Kirt\Desktop\oppo>ls /sdcard/media/0/ 'ls' is not recognized as an internal or external command, operable program or batch file.
> 
> Its `adb shell "ls /sdcard/media/0/"` to relay the command to the watch via adb. Else your windows tries to do `ls`, which is not available on windows.

i want flash directly
but let me backup system first
can guide me to backup the system + boot.img + recovery.img

**Comment by wannaphong** [comment]
@nelikp222 Can you join https://app.element.io/#/room/#Asteroid:matrix.org?

**Comment by nelikp222** [comment]
yes
